app/controllers/api/v0/departments_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class Api::V0::DepartmentsController < Api::V0::BaseController
+
+  before_action :authenticate
+
+  ##
+  # Create a new department based on the information passed in JSON to the API
+  def create
+    unless Api::V0::DepartmentsPolicy.new(@user, nil).index?
+      raise Pundit::NotAuthorizedError
+    end
+    @department = Department.new(org: @user.org,
+                                 code: params[:code],
+                                 name: params[:name])
+    if @department.save
+      redirect_to api_v0_departments_path
+    else
+      # the department did not save
+      self.headers["WWW-Authenticate"] = "Token realm=\"\""
```

Comment:
👍 this great. The v1 json responses contain the error messages but we should probably also list them in the 'realm' as well.